src/background/services/openPayments.ts

```diff
@@ -981,6 +989,66 @@ export class OpenPaymentsService {
     await this.storage.setState({ key_revoked: false });
   }
 
+  async reconnectWallet({ autoKeyAddConsent }: ReconnectWalletPayload) {
+    if (!autoKeyAddConsent) {
+      await this.validateReconnect();
+      return;
+    }
+
+    const { walletAddress } = await this.storage.get(['walletAddress']);
+    if (!walletAddress) {
+      throw new Error('reconnectWallet_error_walletAddressMissing');
+    }
+    if (!KeyAutoAddService.supports(walletAddress)) {
+      throw new ErrorWithKey('connectWalletKeyService_error_notImplemented');
+    }
+
+    try {
+      this.setConnectState('connecting');
+      await this.validateReconnect();
+    } catch (error) {
+      if (isInvalidClientError(error?.cause)) {
```

Comment:
One nit: let's do the inverse error handling in this if, and then rest of this code won't require nesting (early return). Like:
```
if (!isInvalidClientError()) {
  setConnectState(error)
  throw error
}
// Everything else here, without else {}
```